torchvision/models/vgg.py

```diff
@@ -95,107 +95,276 @@ def make_layers(cfg: List[Union[str, int]], batch_norm: bool = False) -> nn.Sequ
 }
 
 
-def _vgg(arch: str, cfg: str, batch_norm: bool, pretrained: bool, progress: bool, **kwargs: Any) -> VGG:
-    if pretrained:
+def _vgg(cfg: str, batch_norm: bool, weights: Optional[WeightsEnum], progress: bool, **kwargs: Any) -> VGG:
+    if weights is not None:
         kwargs["init_weights"] = False
```

Comment:
Proof that the testing strategy works. Prior commit 903a3d5, this optimization was missing from the PR and thus the VGG was being initialized prior loading the weights on the feature branch. Due to this, the set seed was being used for the initialization of the network and thus the input on the pretrained model was different. The issue was caught by the tests that use pretrained expected files and corrected.